### DISCUSSION TITLE
how to share metadata in discovery

### DISCUSSION THREAD

**Discussion by st3b1t** [libp2p/js-libp2p#2589] [Q&A]
I want to use `peerStore.merge`, but I can't seem to get it to work correctly. The goal is to make this new `metadata` field available to all other peers during the `peer:discovery` event.

Here's what I'm trying to do at peer creation time of my peer instance(peerId), want share meta info with other peers:
```javascript
const meta = {my: {meta: 'myinfo'}}
const metaBin = new TextEncoder().encode(JSON.stringify(meta));
await libp2p.peerStore.merge(peerId, {
    metadata: {
      meta: metaBin
    }
})
```
After running this code, I expect the new `metadata` field to be visible to other peers during the `peer:discovery` event, but it doesn't seem to be propagated.

in other peer:
```javascript

libp2p.addEventListener('peer:discovery', async e => {
  const peer = e.detail;
  const peerInfo = await libp2p.peerStore.get(peer.id);
  console.log('peerInfo:',peerInfo)
```
peerInfo contains empty Map(0) metadata, this is console.log outout:
```js
peerInfo: {
  addresses: [],
  protocols: [],
  metadata: Map(0) {},
  tags: Map(1) { 'bootstrap' => { value: 50, expiry: 1718117050637n } },
  id: PeerId(QmckJMAv2BfATF1FjcqPndJHoWKbqACNoEvmsWN8unxyiG),
  peerRecordEnvelope: undefined
}
```

Has anyone experienced something similar or have any suggestions on how to resolve this issue? Any advice would be greatly appreciated!

**Comment by achingbrain** [comment]
The `peer:discovery` event detail type is [PeerInfo](https://docs.libp2p.io/concepts/fundamentals/peers/#peer-info) which doesn't include metadata.

The updated peer data should be available via a `peer:update` event after the merge - have you tried that?

**Reply by st3b1t** [reply]
please, where is the documentation about it `peer:update`? 
It's a bit difficult to get around developing over this lib, unfortunately the docs about it is not very clear
for example a list of events about a PeerInfo

**Reply by achingbrain** [reply]
The various events libp2p will emit are typed and documented in the API docs for the [Libp2pEvents type](https://libp2p.github.io/js-libp2p/interfaces/_libp2p_interface.Libp2pEvents.html).
